The search should be not ok when dimension in search vector different from that in insert vectors

**Please state your issue using the following template and, most importantly, in English.**

**Describe the bug**
The test case is shown below. The dim in query vectors is 64, and in insert vectors is 128.
```
public void  testSearchVectorNotExisted(MilvusClient client, String collectionName) {
        InsertParam insertParam = new InsertParam.Builder(collectionName).withFields(Constants.defaultEntities).build();
        InsertResponse res = client.insert(insertParam);
        assert(res.getResponse().ok());
        client.flush(collectionName);
        List<List<Float>> query = Utils.genVectors(nq,64, false);
        String dsl = String.format(
                "{\"bool\": {"
                        + "\"must\": [{"
                        + "    \"range\": {"
                        + "        \"int64\": {\"GT\": -10, \"LT\": 1000}"
                        + "    }},{"
                        + "    \"vector\": {"
                        + "        \"float_vector\": {"
                        + "            \"topk\": %d, \"metric_type\": \"L2\", \"type\": \"float\", \"query\": %s, \"params\": {\"nprobe\": 20}"
                        + "    }}}]}}",
                top_k, query);
        SearchParam searchParam = new SearchParam.Builder(collectionName).withDSL(dsl).build();
        SearchResponse resSearch = client.search(searchParam);
        System.out.println(resSearch.getFieldsMap().size());
        List<List<Map<String, Object>>> fieldsMap = resSearch.getFieldsMap();
        fieldsMap.stream().forEach(item-> System.out.println(item.toString()));
        Assert.assertFalse(resSearch.getResponse().ok());
```
The console output is:
```
INFO: Search completed successfully! Returned results for 2 queries
2
[{}, {}, {}, {}, {}, {}, {}, {}, {}, {}]
[{}, {}, {}, {}, {}, {}, {}, {}, {}, {}]

java.lang.AssertionError: expected [false] but found [true]
Expected :false
Actual   :true
```
**Steps/Code to reproduce behavior**
Follow this [guide](http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) to craft a minimal bug report. This helps us reproduce the issue you're having and resolve the issue more quickly.

**Expected behavior**
A clear and concise description of what you expected to happen.

**Method of installation**
- [ ] Docker/cpu
- [x] Docker/gpu
- [ ] Build from source

**Environment details**
- Hardware/Softwars conditions (OS, CPU, GPU, Memory)


- Milvus version (master or released version)


**Configuration file**
Settings you made in `server_config.yaml` or `milvus.yaml`

```yaml
paste-file-content-here
```

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

--- a/core/src/query/GeneralQuery.h
+++ b/core/src/query/GeneralQuery.h
@@ -70,6 +70,7 @@ struct RangeQuery {
 using RangeQueryPtr = std::shared_ptr<RangeQuery>;
 
 struct VectorRecord {
+    size_t vector_count;
     std::vector<float> float_data;
     std::vector<uint8_t> binary_data;
 };

--- a/core/src/server/delivery/request/SearchReq.cpp
+++ b/core/src/server/delivery/request/SearchReq.cpp
@@ -69,8 +69,6 @@ SearchReq::OnExecute() {
             }
         }
 
-        int64_t dimension = 0;
-
         // step 4: Get field info
         std::unordered_map<std::string, engine::DataType> field_types;
         for (auto& schema : fields_schema) {
@@ -78,7 +76,21 @@ SearchReq::OnExecute() {
             field_types.insert(std::make_pair(field->GetName(), field->GetFtype()));
             if (field->GetFtype() == engine::DataType::VECTOR_FLOAT ||
                 field->GetFtype() == engine::DataType::VECTOR_BINARY) {
-                dimension = field->GetParams()[engine::PARAM_DIMENSION];
+                // check dim
+                int64_t dimension = field->GetParams()[engine::PARAM_DIMENSION];
+                auto vector_query = query_ptr_->vectors.begin()->second;
+                if (!vector_query->query_vector.binary_data.empty()) {
+                    if (vector_query->query_vector.binary_data.size() !=
+                        vector_query->query_vector.vector_count * dimension / 8) {
+                        return Status(SERVER_INVALID_ARGUMENT, "query vector dim not match");
+                    }
+                } else if (!vector_query->query_vector.float_data.empty()) {
+                    if (vector_query->query_vector.float_data.size() !=
+                        vector_query->query_vector.vector_count * dimension) {
+                        return Status(SERVER_INVALID_ARGUMENT, "query vector dim not match");
+                    }
+                }
+
                 // validate search metric type and DataType match
                 bool is_binary = (field->GetFtype() == engine::DataType::VECTOR_FLOAT) ? false : true;
                 if (query_ptr_->metric_types.find(field->GetName()) != query_ptr_->metric_types.end()) {
@@ -126,11 +138,7 @@ SearchReq::OnExecute() {
         fiu_do_on("SearchReq.OnExecute.empty_result_ids", result_->result_ids_.clear());
         if (result_->result_ids_.empty()) {
             auto vector_query = query_ptr_->vectors.begin()->second;
-            if (!vector_query->query_vector.binary_data.empty()) {
-                result_->row_num_ = vector_query->query_vector.binary_data.size() * 8 / dimension;
-            } else if (!vector_query->query_vector.float_data.empty()) {
-                result_->row_num_ = vector_query->query_vector.float_data.size() / dimension;
-            }
+            result_->row_num_ = vector_query->query_vector.vector_count;
             return Status::OK();  // empty table
         }
 

--- a/core/src/server/grpc_impl/GrpcRequestHandler.cpp
+++ b/core/src/server/grpc_impl/GrpcRequestHandler.cpp
@@ -217,8 +217,9 @@ DeSerialization(const ::milvus::grpc::GeneralQuery& general_query, query::Boolea
                     CopyRowRecords(query.vector_query().records(),
                                    google::protobuf::RepeatedField<google::protobuf::int64>(), vectors);
 
-                    vector_query->query_vector.float_data = vectors.float_data_;
-                    vector_query->query_vector.binary_data = vectors.binary_data_;
+                    vector_query->query_vector.vector_count = vectors.vector_count_;
+                    vector_query->query_vector.float_data.swap(vectors.float_data_);
+                    vector_query->query_vector.binary_data.swap(vectors.binary_data_);
 
                     vector_query->boost = query.vector_query().query_boost();
                     vector_query->field_name = query.vector_query().field_name();
@@ -1733,8 +1734,9 @@ GrpcRequestHandler::DeserializeJsonToBoolQuery(
             engine::VectorsData vector_data;
             CopyRowRecords(vector_param.row_record().records(),
                            google::protobuf::RepeatedField<google::protobuf::int64>(), vector_data);
-            vector_query->query_vector.binary_data = vector_data.binary_data_;
-            vector_query->query_vector.float_data = vector_data.float_data_;
+            vector_query->query_vector.vector_count = vector_data.vector_count_;
+            vector_query->query_vector.binary_data.swap(vector_data.binary_data_);
+            vector_query->query_vector.float_data.swap(vector_data.float_data_);
 
             query_ptr->vectors.insert(std::make_pair(placeholder, vector_query));
         }

--- a/core/src/server/web_impl/handler/WebRequestHandler.cpp
+++ b/core/src/server/web_impl/handler/WebRequestHandler.cpp
@@ -636,12 +636,15 @@ WebRequestHandler::ProcessLeafQueryJson(const nlohmann::json& json, milvus::quer
             if (param_json.contains("metric_type")) {
                 std::string metric_type = param_json["metric_type"];
                 vector_query->metric_type = metric_type;
-                query_ptr->metric_types.insert({vector_name, param_json["metric_type"]});
+                query_ptr->metric_types.insert({vector_name, metric_type});
             }
             if (!vector_param_it.value()["params"].empty()) {
                 vector_query->extra_params = vector_param_it.value()["params"];
             }
-            for (auto& vector_records : vector_param_it.value()["values"]) {
+
+            auto& values = vector_param_it.value()["values"];
+            vector_query->query_vector.vector_count = values.size();
+            for (auto& vector_records : values) {
                 if (field_type_.find(vector_name) != field_type_.end()) {
                     if (field_type_.at(vector_name) == engine::DataType::VECTOR_FLOAT) {
                         for (auto& data : vector_records) {

